test/provider/ContextHandler.unittest.ts

```diff
@@ -25,7 +25,15 @@ let mockGetChannelMembers: jest.Mock<AppWindow[], [ContextChannel]>;
 let mockGetWindowsListeningToChannel: jest.Mock<AppWindow[], [ContextChannel]>;
 
 function createMockAppWindowWithName(name: string): AppWindow {
-    return {...createMockAppWindow(), identity: {uuid: 'test', name}};
+    const mockedWindow = createMockAppWindow();
+    mockedWindow.isReadyToReceiveContext.mockResolvedValue(true);
+    return {...mockedWindow, identity: {uuid: 'test', name}};
+}
+
+function createMockAppWindowNotReadyWithName(name: string): AppWindow {
+    const mockedWindow = createMockAppWindow();
```

Comment:
Can `createMockAppWindow()` be changed to take a `Partial<AppWindow>` and just:
```ts
export function createMockAppWindow(options: Partial<AppWindow> = {}): jest.Mocked<AppWindow> {
    return {
        id: '',
        // existing...
        ...options
    };
}
```

Then we wont need to create all these types of functions.

Reply:
I'd rather not. This mock is used in instances where we need to pass an `AppWindow` into some dependent module. There is no guarantee that it will only ever use those properties, so I think it makes sense to expose a fully mocked version that implements the whole interface.

Reply:
`// existing...` was meant to be shortcut for what is currently there in `createMockAppWindow`. It would be the same as it is now, but allow overrides for to be passed in via `options`.

Reply:
I may have misread this a bit. Will circle back in a bit once I'm clear what's being asked.